### PR TITLE
[#2799] Fix authz.auth_is_anon_user() when couldn't get the user's IP

### PR DESCRIPTION
**Pull request by vitorbaptista**
Also improve the method for checking if the username is an IP.

### CODE REVIEW

File: ckan/authz.py [hunk 1]
```diff
@@ -430,12 +430,15 @@ def auth_is_anon_user(context):
 
         See ckan/lib/base.py:232 for pylons context object logic
     '''
+    is_ip = lambda x: re.match('\d+\.\d+\.\d+\.\d+', x)
+    is_unknown_ip = lambda x: x == 'Unknown IP Address'
```
Comment (davidread) [inline ckan/authz.py]: This magic value comes from https://github.com/ckan/ckan/blob/master/ckan/lib/base.py#L261 so I'd encourage you to reference it in a comment.

**Comment by davidread** [comment]
Otherwise, this looks great.

**Comment by wardi** [comment]
I feel strongly that we should remove the madness of mixing IP addresses and user names in the same variable instead of further cementing this behaviour into core.

**Comment by wardi** [comment]
I'll work on a PR that replaces all the `'user': c.user or c.author` code with `'user': c.user` in our controllers

**Comment by davidread** [comment]
Good plan. The IP is still always available in the request object if we should ever want it.

**Comment by amercader** [comment]
This is probably superseded by #2817

**Comment by amercader** [comment]
No longer relevant after #2817

**Pull request closed without merging**